enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,1139 @@
+// VSpherePlatformNodeNetworkingSpec holds the network cidrs for
+// including and excluding in the cloud provider.
+type VSpherePlatformNodeNetworkingSpec struct {
+
+  // IP address on VirtualMachine's network interfaces included in the fields' CIDRs
+  // that will be used in respective status.addresses fields.
+  // +kubebuilder:validation:Optional
+  NetworkSubnetCIDR string `json:"networkSubnetCidr,omitempty"`
+
+  // IP address on VirtualMachine's VM Network names that will be used to when searching
+  // for status.addresses fields. Note that if InternalNetworkSubnetCIDR and
+  // ExternalNetworkSubnetCIDR are not set, then the vNIC associated to this network must
+  // only have a single IP address assigned to it.
+  // +kubebuilder:validation:Optional
+  Network string `json:"network,omitempty"`
+
+  // IP addresses in these subnet ranges will be excluded when selecting
+  // the IP address from the VirtualMachine's VM for use in the
+  // status.addresses fields.
+  // +kubebuilder:validation:Optional
+  ExcludeNetworkSubnetCIDR string `json:"excludeNetworkSubnetCidr,omitempty"`
+}
+
+type VSpherePlatformNodeNetworking struct {
+  External VSpherePlatfornNodeNetworkingSpec `json:"external"`
+  Internal VSpherePlatfornNodeNetworkingSpec `json:"internal"`
+}
+
+// VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type VSpherePlatformSpec struct {
+  // vcenters holds the connection details for services to communicate with vCenter.
+  // Currently only a single vCenter is supported.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MaxItems=1
+  // +kubebuilder:validation:MinItems=1
+  VCenters []VSpherePlatformVCenterSpec `json:"vcenters,omitempty"`
+
+  // deploymentZones holds the association between vcenter, failure domains
+  // and vcenter placement for virtual machines.
+  // +kubebuilder:validation:Optional
+  DeploymentZones []VSpherePlatformDeploymentZoneSpec `json:"deploymentZones,omitempty"`
+
+  // failureDomains holds the VSpherePlatformFailureDomainSpec which contains
+  // the definition of region, zone and the vCenter topology.
+  // If this is omitted failure domains (regions and zones) will not be used.
+  // +kubebuilder:validation:Optional
+  FailureDomains []VSpherePlatformFailureDomainSpec `json:"failureDomains,omitempty"`
+
+  // nodeNetworking holds the VSpherePlatformNodeNetworking which contains
+  // the definition of ...
+  // If this field is omitted the default mechanism for determining
+  // which ip addresses to be used - something...
+  // +kubebuilder:validation:Optional
+  NodeNetworking VSpherePlatformNodeNetworking `json:"nodeNetworking,omitempty"`
+}
+```
+
+### CCCMO
+
+The CCCMO will need to be modified to support the VSpherePlatformSpec.
+It will need to be able to use the existing INI-based cloud-config spec
+and also translate that configuration to YAML-based cloud-config.
```

Comment:
is it worth noting that the cccmo will need to be modified to take the new values from the infrastructure config and add them to the vsphere cloud config?